src/client/Microsoft.Identity.Client/Platforms/Features/WinFormsLegacyWebUi/CustomWebBrowser.CustomWebBrowserEvent.cs

```diff
@@ -33,27 +33,19 @@ public CustomWebBrowserEvent(CustomWebBrowser parent)
                 cancel = e.Cancel;
             }
 
-            // This method are empty because we agree with their implementation in base class event handler System.Windows.Forms.WebBrowser+WebBrowserEvent.
-            // We disagree with empty implementation of NavigateError.
-            // We also could tweak implementation of base class if we disagree by implementing this method.
-            // This is COM events handler, defined in COM interface, however this model works as Events in .NET
-            // Multiple handlers are possible, so empty method just called and do nothing.
-            public void BeforeNavigate2(object pDisp, ref object urlObject, ref object flags, ref object targetFrameName,
+            public void BeforeNavigate2(object pDisp, ref object url, ref object flags, ref object targetFrameName,
```

Comment:
Using `BeforeNavigate2` instead of `Navigating` because it exposes post data. But `BeforeNavigate2` doesn't have a default .NET wrapper, so similar to `NavigateError`, have to create our own wrapper, event handlers, event args.

Reply:
Why is it called `BeforeNavigate2`? Is that the official .NET name?

Reply:
Yes.